### DISCUSSION TITLE
An html-keyboard-response plugin that does not erase the screen?

### DISCUSSION THREAD

**Discussion by dcousin3** [jspsych/jsPsych#923] [Q&A] [answered]
It would be convenient to have a plugin that can add some html code to the display without erasing what is currently on the display.

I have tried modifying the current html-keyboard-response plugin by 
a) removing the instruction `display_element.innerHTML = '';`
b) adding the new html instructions to the existing div with 
`display_element.innerHTML += new_html;`

but that does not seems to be working. Any help appreciated.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @dcousin3 ,

Can you say a little bit more about the use case that you are imagining here? There are some other methods for having persisting elements on the screen, like [specifying which element is the display_element](https://www.jspsych.org/core_library/jspsych-core/#jspsychinit). 

I agree that sometimes it would be easier to setup my experiments if the display were modifiable instead of replaceable, but it's also hard for me to come up with a plugin design that would work in the general case. Happy to hear ideas on this!

**Reply by dcousin3** [reply]
I can see many applications. For example:
1-Adding a feedback below the stimuli so that the participant can look back and guess why he/she made a mistake;

2-To show many items on a single display: Granted, all the items could be listed in a single "stimulus" attribute, but for modularity, it would be easier to have a timeline and timeline_variables calling the plugin "ShowOneItem" many times. In E-Prime, this was used frequently as it is possible to set the background of a slide "transparent". Hence, the subsequent slides would show up above the previous ones.

**Reply by dcousin3** [reply]
Hello,

In the attached zip, I included a plugin html-keyboard-response-noerase that adds additional html to the existing `<div>`. 

I am not sure how bullet-proof this is, but seems to be doing the work? Thanks for the cue regarding `jsPsych.getDisplayElement()`.

[jspsych-html-keyboard-response-noerase.zip](https://github.com/jspsych/jsPsych/files/4983260/jspsych-html-keyboard-response-noerase.zip)

Denis.
